optuna/storages/_rdb/alembic/versions/v3.0.0.a.py

```diff
@@ -0,0 +1,216 @@
+        new_distribution = FloatDistribution(
+            low=distribution.low,
+            high=distribution.high,
+            log=False,
+            step=distribution.q,
+        )
+
+    # Integer distributions.
+    elif isinstance(distribution, IntUniformDistribution):
+        new_distribution = IntDistribution(
+            low=distribution.low,
+            high=distribution.high,
+            log=False,
+            step=distribution.step,
+        )
+    elif isinstance(distribution, IntLogUniformDistribution):
+        new_distribution = IntDistribution(
+            low=distribution.low,
+            high=distribution.high,
+            log=True,
+            step=distribution.step,
+        )
+
+    # Categorical distribution.
+    else:
+        new_distribution = distribution
+
+    return distribution_to_json(new_distribution)
+
+
+def restore_old_distribution(distribution_json: str) -> str:
+    distribution = json_to_distribution(distribution_json)
+    old_distribution: BaseDistribution
+
+    # Float distributions.
+    if isinstance(distribution, FloatDistribution):
+        if distribution.log:
+            old_distribution = LogUniformDistribution(
+                low=distribution.low,
+                high=distribution.high,
+            )
+        else:
+            if distribution.step is not None:
+                old_distribution = DiscreteUniformDistribution(
+                    low=distribution.low,
+                    high=distribution.high,
+                    q=distribution.step,
+                )
+            else:
+                old_distribution = UniformDistribution(
+                    low=distribution.low,
+                    high=distribution.high,
+                )
+
+    # Integer distributions.
+    elif isinstance(distribution, IntDistribution):
+        if distribution.log:
+            old_distribution = IntLogUniformDistribution(
+                low=distribution.low,
+                high=distribution.high,
+                step=distribution.step,
+            )
+        else:
+            old_distribution = IntUniformDistribution(
+                low=distribution.low,
+                high=distribution.high,
+                step=distribution.step,
+            )
+
+    # Categorical distribution.
+    else:
+        old_distribution = distribution
+
+    return distribution_to_json(old_distribution)
+
+
+def upgrade() -> None:
+    bind = op.get_bind()
+    inspector = Inspector.from_engine(bind)
+    tables = inspector.get_table_names()
+
+    assert "trial_params" in tables
+
+    session = orm.Session(bind=bind)
+    try:
+        distributions = session.query(TrialParamModel).all()
+        for distribution in distributions:
+            distribution.distribution_json = migrate_new_distribution(
+                distribution.distribution_json,
+            )
+        session.bulk_save_objects(distributions)
+        session.commit()
```

Comment:
@himkt mentioned that we could divide batches for the users who have millions of trials.

Reply:
✅  in 6c2df8c.

Reply:
Thanks!